### PR TITLE
Port Misc InteropServices APIs

### PR DESCRIPTION
**Pull request by shrah**
Port  the following miscellenous InteropServices API Changes:

M:System.Runtime.InteropServices.COMException.ToString
M:System.Runtime.InteropServices.CriticalHandle.Close
M:System.Runtime.InteropServices.HandleRef.op_Explicit(System.Runtime.InteropServices.HandleRef)~System.IntPtr
M:System.Runtime.InteropServices.HandleRef.ToIntPtr(System.Runtime.InteropServices.HandleRef)

### CODE REVIEW

**Comment by shrah** [comment]
@yzha @tijoytom @jkotas @weshaggard

**Comment by weshaggard** [comment]
The API addition looks correct, but we should add tests for these APIs.

**Comment by jkotas** [comment]
@shrah Could you please sign the CLA at https://cla2.dotnetfoundation.org/ ?

File: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs [hunk 1]
```diff
@@ -231,6 +231,7 @@ public partial class COMException : System.Exception
         public COMException(string message) { }
         public COMException(string message, System.Exception inner) { }
         public COMException(string message, int errorCode) { }
+        public override string ToString() { return default(string); }
```
Comment (justinvp) [inline src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs]: Nit: `throw null;` ?

**Comment by shrah** [comment]
@jkotas, I've signed the CLA.

@weshaggard I'll add some tests.

File: src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs]: As long as you're adding this, can you add the ```protected COMException(SerializationInfo info, StreamingContext context)``` ctor as well and change the type to derive from ExternalException?

Comment (shrah) [inline src/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs]: Sure, I'll add those too.

**Comment by stephentoub** [comment]
This looks like a strict subset of https://github.com/dotnet/corefx/pull/12873.  Thanks for doing this, but I'm going to close this one so we can merge the more comprehensive one without conflicts.

**Pull request closed without merging**